os/kernel/binary_manager/binary_manager_data.c

```diff
@@ -367,6 +387,16 @@ int binary_manager_check_user_update(int bin_idx)
 
 	bmvdbg("Checking [%d] current version %d, part_idx %d\n", bin_idx, running_ver, part_idx);
 
+#ifdef CONFIG_BINARY_SIGNING
+	/* Check signature */
+	ret = up_verify_usersignature(BIN_PARTADDR(bin_idx, part_idx));
+	if (ret == OK) {
+		bmvdbg("%s Signature Checking Success\n", BIN_NAME(bin_idx));
+	} else {
+		bmdbg("Invalid Signature, name : %s, address : %p\n", BIN_NAME(bin_idx), BIN_PARTADDR(bin_idx, (BIN_USEIDX(bin_idx))));
+		return BINMGR_NOT_FOUND;
```

Comment:
may be new error code can be added for sign check?

Reply:
We don't distinguish signature error and other invalid binary errors like crc now.
I will update later.